packages/material-ui/src/utils/useControlled.js

```diff
@@ -1,7 +1,7 @@
 /* eslint-disable react-hooks/rules-of-hooks, react-hooks/exhaustive-deps */
 import React from 'react';
 
-export default function useControlled({ controlled, default: defaultProp, name }) {
+export default function useControlled({ controlled, default: defaultProp, name, propHint }) {
```

Comment:
We recently had a discussion that highlighted a potential source of confusion from this warning.
The Autocomplete has two states that can be controlled. It's not the component itself that is controlled but the state of the component.

What do you think of delaying this propHint to another pull request? There are two other components that could leverage it:
- Autocomplete value and inputValue states 
- SelectInput value and open states

In the above two cases, I think that useControlled should be used twice, not once like currently.

Note that the DataGrid will have a couple of states; I'm at least aware of page, filter, and sorting.